app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt

```diff
@@ -99,18 +98,18 @@ class MainViewModel : ViewModel() {
     private suspend fun fetchData() = withContext(Dispatchers.Default) {
         try {
             withContext(Dispatchers.IO) {
-                response = ServiceFactory.createRetrofitService(
+                val response = ServiceFactory.createRetrofitService(
                     CurrencyApi::class.java,
                     "http://api.evp.lt/currency/commercial/exchange/"
                 )
                     .getCurrencyAsync(url).await()
-            }
 
-            if (response!!.body() != null) {
-                calculateValues()
-                makeInfoMessage()
-            } else {
-                _error.postValue(Unit)
+                if (response.body() != null) {
+                    calculateValues(response.body())
+                    makeInfoMessage(response.body())
+                } else {
+                    _error.postValue(Unit)
+                }
```

Comment:
Replace this with: 
`           response.body()?.let{body ->
                    calculateValues(body)
                    makeInfoMessage(body)
                }
                if (response.body() == null) {
                    _error.postValue(Unit)
                }`